src/Features/Core/Portable/AddImport/CodeActions/AddImportCodeAction.cs

```diff
@@ -9,66 +9,61 @@
+/// metadata reference.  We don't use the standard code action types because
+/// we want to do things like show a glyph if this will do more than just add
+/// an import.
+/// </summary>
+internal abstract class AddImportCodeAction : CodeAction
 {
-    internal abstract partial class AbstractAddImportFeatureService<TSimpleNameSyntax>
-    {
-        /// <summary>
-        /// Code action we use when just adding a using, possibly with a project or
-        /// metadata reference.  We don't use the standard code action types because
-        /// we want to do things like show a glyph if this will do more than just add
-        /// an import.
-        /// </summary>
-        private abstract class AddImportCodeAction : CodeAction
-        {
-            protected readonly AddImportFixData FixData;
-            private readonly CodeActionPriority _priority;
+    protected readonly AddImportFixData FixData;
+    private readonly CodeActionPriority _priority;
 
-            public override string Title { get; }
-            public sealed override ImmutableArray<string> Tags { get; }
+    public override string Title { get; }
+    public sealed override ImmutableArray<string> Tags { get; }
 
-            public sealed override string EquivalenceKey => Title;
+    public sealed override string EquivalenceKey => Title;
 
-            /// <summary>
-            /// The <see cref="Document"/> we started the add-import analysis in.
-            /// </summary>
-            protected readonly Document OriginalDocument;
+    /// <summary>
+    /// The <see cref="Document"/> we started the add-import analysis in.
+    /// </summary>
+    protected readonly Document OriginalDocument;
 
-            /// <summary>
-            /// The changes to make to <see cref="OriginalDocument"/> to add the import.
-            /// </summary>
-            private readonly ImmutableArray<TextChange> _textChanges;
+    /// <summary>
+    /// The changes to make to <see cref="OriginalDocument"/> to add the import.
+    /// </summary>
+    private readonly ImmutableArray<TextChange> _textChanges;
 
-            protected AddImportCodeAction(
-                Document originalDocument,
-                AddImportFixData fixData,
-                ImmutableArray<string> additionalTags)
-            {
-                OriginalDocument = originalDocument;
-                FixData = fixData;
+    protected AddImportCodeAction(
+        Document originalDocument,
+        AddImportFixData fixData,
+        ImmutableArray<string> additionalTags)
+    {
+        OriginalDocument = originalDocument;
+        FixData = fixData;
 
-                Title = fixData.Title;
-                Tags = fixData.Tags.ToImmutableArrayOrEmpty().AddRange(additionalTags);
-                _priority = fixData.Priority;
-                _textChanges = fixData.TextChanges.ToImmutableArrayOrEmpty();
+        Title = fixData.Title;
+        Tags = fixData.Tags.AddRange(additionalTags);
```

Comment:
>  [](http://example.com/codeflow?start=2&length=1)

I think the "ToImmutableArrayOrEmpty()" is needed since tags default to `default`
[https://github.com/dotnet/roslyn/blob/3e516fcb5478235ed85fec40803fa3676fa8ff4d/src/Features/Core/Portable/AddImport/AddImportFixData.cs#L21C32-L21C37](https://github.com/dotnet/roslyn/blob/3e516fcb5478235ed85fec40803fa3676fa8ff4d/src/Features/Core/Portable/AddImport/AddImportFixData.cs#L21C32-L21C37)

Reply:
Fair :)